src/useHotkeys.ts

```diff
@@ -143,7 +143,7 @@ export default function useHotkeys<T extends HTMLElement>(
         parseKeysHookInput(keys, memoisedOptions?.splitKey).forEach((key) => proxy.removeHotkey(parseHotkey(key, memoisedOptions?.combinationKey)))
       }
     }
-  }, [keys, memoisedOptions, enabledScopes])
+  }, [JSON.stringify(keys), memoisedOptions, enabledScopes])
```

Comment:
Hmh... I don't really want to use JSON.stringify here. It's an expensive operation and you shouldn't hack the dependency array that way. A better approach in my opinion would be to make sure we are always dealing with a string:

```ts
const _keys = keys instanceof Array ? keys.join(_options?.splitKey) : keys
```

Then we can reference the `_keys` everywhere we use `keys` and also the check in `parseKeysHookInput` could be removed.